Sources/Extensions/CoreGraphics/CGSizeExtensions.swift

```diff
@@ -50,5 +50,142 @@ public extension CGSize {
 		return CGSize(width: aWidth, height: aHeight)
 	}
 
+}
+
+// MARK: - Operators
+public extension CGSize {
+
+    /// SwifterSwift: Add two CGSize
+    ///
+    ///     let sizeA = CGSize(width: 5, height: 10)
+    ///     let sizeB = CGSize(width: 3, height: 4)
+    ///     let result = sizeA + sizeB
+    ///     //result = CGSize(width: 8, height: 14)
+    ///
+    /// - Parameters:
+    ///   - lhs: CGSize to add to.
+    ///   - rhs: CGSize to add.
+    /// - Returns: The result comes from the addition of the two given CGSize struct.
+    public static func + (lhs: CGSize, rhs: CGSize) -> CGSize {
+        return CGSize(width: lhs.width + rhs.width, height: lhs.height + rhs.height)
+    }
+
+    /// SwifterSwift: Add a CGSize to self.
+    ///
+    ///     let sizeA = CGSize(width: 5, height: 10)
+    ///     let sizeB = CGSize(width: 3, height: 4)
+    ///     sizeA += sizeB
+    ///     //sizeA = CGPoint(width: 8, height: 14)
+    ///
+    /// - Parameters:
+    ///   - lhs: self
+    ///   - rhs: CGSize to add.
+    public static func += (lhs: inout CGSize, rhs: CGSize) {
+        lhs.width += rhs.width
+        lhs.height += rhs.height
+    }
+
+    /// SwifterSwift: Subtract two CGSize
+    ///
+    ///     let sizeA = CGSize(width: 5, height: 10)
+    ///     let sizeB = CGSize(width: 3, height: 4)
+    ///     let result = sizeA - sizeB
+    ///     //result = CGSize(width: 2, height: 6)
+    ///
+    /// - Parameters:
+    ///   - lhs: CGSize to subtract from.
+    ///   - rhs: CGSize to subtract.
+    /// - Returns: The result comes from the subtract of the two given CGSize struct.
+    public static func - (lhs: CGSize, rhs: CGSize) -> CGSize {
+        return CGSize(width: lhs.width - rhs.width, height: lhs.height - rhs.height)
+    }
+
+    /// SwifterSwift: Subtract a CGSize from self.
+    ///
+    ///     let sizeA = CGSize(width: 5, height: 10)
+    ///     let sizeB = CGSize(width: 3, height: 4)
+    ///     sizeA -= sizeB
+    ///     //sizeA = CGPoint(width: 2, height: 6)
+    ///
+    /// - Parameters:
+    ///   - lhs: self
+    ///   - rhs: CGSize to subtract.
+    public static func -= (lhs: inout CGSize, rhs: CGSize) {
+        lhs.width -= rhs.width
+        lhs.height -= rhs.height
+    }
+
+    /// SwifterSwift: Multiply two CGSize
+    ///
+    ///     let sizeA = CGSize(width: 5, height: 10)
+    ///     let sizeB = CGSize(width: 3, height: 4)
+    ///     let result = sizeA * sizeB
+    ///     //result = CGSize(width: 15, height: 40)
+    ///
+    /// - Parameters:
+    ///   - lhs: CGSize to multiply.
+    ///   - rhs: CGSize to multiply with.
+    /// - Returns: The result comes from the multiplication of the two given CGSize structs.
+    public static func * (lhs: CGSize, rhs: CGSize) -> CGSize {
```

Comment:
I don't understand the idea of multiplying 2 sizes. Multiplying by a scalar makes perfect sense, but when will you ever multiply sizes? I'm just curious to know of any examples.

Reply:
According to the docs (https://developer.apple.com/documentation/coregraphics/cgsize), CGSize struct may be used to represent a distance vector. The idea behind multiplying sizes comes from multiplying those vectors.

I have to admit i have never used CGSizes for this purpose before.

You may want to manipulate the size of a view's frame defined by a vector and not a scalar value (specially when you transform its size not proportionally).

I suppose a use-case like this below:
```swift
let someVector: CGSize = CGsize(width: 1.5, height: 4)
aView.frame.size *= someVector
```

In this case someVector is a type of CGSize but represents a vector rather than a size.

@guykogus what do you think? Am i wrong in this situation? Is there a better solution for this?

Reply:
I think that if you want a vector you'd be better off using `CGVector`, no?
But maybe I can see what you mean, like if you have an image that's scaled (losing aspect ratio) to fit a view, and you want to get the ratios you could maybe do `image.size / view.frame.size` and then use that size to multiply something connected.

Reply:
I agree a vector is better to be represented as a CGVector rather than a CGSize.

I realize that in my last example it would be a better fit to multiply the size with a CGVector. Then we should add an operator for multiply a cgsize with a cgvector instead of cgsize with cgsize. Shall i? :)

Reply:
No, I think it's fine to leave it with `CGSize`.